File: downloads/code/LeetCode/Python/Zigzag-Conversion.py
class Solution:
    # @param {string} s
    # @param {integer} numRows
    # @return {string}
    def convert(self, s, numRows):
        if numRows <= 1 or numRows >=len(s):
            return s

        zigzag = ["" for i in range(numRows)]
        step = 1
        i = 0
        for c in s:
            zigzag[i] += c
            # 从0行开始的话往下走
            if i == 0:
                step = 1
            # 到达末尾行要返回
            if i == numRows -1:
                step = -1
            i += step
        ret = ""
        for i in zigzag:
            ret += i
        return ret

File: downloads/code/LeetCode/Java/ZigZag-Conversion.java
public class Solution {
    public String convert(String s, int numRows) {
        char[] c = s.toCharArray();
        int len = c.length;
        StringBuffer[] sb = new StringBuffer[numRows];
        for (int i=0;i<numRows;i++)
            sb[i] = new StringBuffer();
        int i =0;
        while(i<len){
            for(int idx = 0; idx<numRows && i<len; idx++)
                sb[idx].append(c[i++]);
            for(int idx = numRows-2; idx>=1 && i<len; idx--)
                sb[idx].append(c[i++]);
        }

        for(i=1;i<numRows;i++)
            sb[0].append(sb[i]);
        return sb[0].toString();
    }
}
